### ISSUE TITLE
Header only Pods producing "file not found" errors 

### ISSUE THREAD

**Issue by HeEAaD** [Ruenzuo/cocoapods-amimono#10]
# Issue
Patching Pods with Amimono which having neither .m or .swift files will produce "file not found" errors during the the linker phase.

The problem is in the `[Amimono] Create filelist per architecture` build phase. We should check if the o-file exists before adding it to the `filelist`.

# Steps to reproduce
1. add Amimono and a Pod with only *.h files (e.g. CWLSynthesizeSingleton) in the Podfile (see the example project: 
[HeaderFileOnly Example.zip](https://github.com/Ruenzuo/cocoapods-amimono/files/430962/HeaderFileOnly.Example.zip)
 )
2. pod install
3. open the generated workspace
4. build the project
5. You will see missing file errors during the linker phase

# Output from the "HeaderFileOnly Example"
```
ld: file not found: /Users/steffen.matthischke/Library/Developer/Xcode/DerivedData/HeaderFileOnly_Example-alnxwqcjolmgzjgzerfcvyfnynrs/Build/Intermediates/Pods.build/Debug-iphonesimulator/CWLSynthesizeSingleton.build/Objects-normal/x86_64/*.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

# Environment
cocoapods-amimono 0.0.3
cocoapods 1.0.1
Xcode 7.3.1

**Comment by Ruenzuo** [comment]
[This part of the script](https://github.com/Ruenzuo/cocoapods-amimono/blob/master/lib/cocoapods-amimono/integrator.rb#L10-L14) should only write to `filelist` if there is something found, I was sort of expecting bash to handle this gracefully but I was obviously wrong. I will have a look at this soon!

**Comment by Ruenzuo** [comment]
@HeEAaD Thank you for reporting this issue. This should be fix here: https://github.com/Ruenzuo/cocoapods-amimono/pull/11, could you verify that it works with the considerations mentioned in the PR?

**Comment by HeEAaD** [comment]
@Ruenzuo #11 fixes this Issue if the pod without implementation files is not the only pod in the workspace (e.g. "CWLSynthesizeSingleton" and "pop") 👍 

If the pod without implementation files is the only pod in the workspace than there is a build error: 
```
ld: file not found: 
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
This is ok in my eyes.

**Comment by Ruenzuo** [comment]
@HeEAaD yes, that's the edge case I mentioned in the PR. It's OK for me as well, as people should only use this plugin if either they have many dependencies in the Podfile or they have many transitive dependencies, having a single Pod without sources and many transitive dependencies is very unlikely so I wouldn't optimise for that case for now.

**Comment by mvyrko** [comment]
Hi @Ruenzuo,
I have about 15 pods in project and have error 
```
ld: file not found: 
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
How can I fix it? 

Thank you in advance

**Comment by Ruenzuo** [comment]
I'm happy to help if you could provide a Podfile and an empty project to reproduce the error on my side.

Another thing: 15 pods doesn't sound like a good reason to use this plugin for me (we have over 50 pods, that's why we use it). But maybe you still see additional value in it (load time, all resources on main bundle), so just wanted to point that out.

**Comment by mvyrko** [comment]
Hi,
Unfortunately, I can't provide my Podfile and empty project.
I need to decrease app launch time on device below 9.3. 
So, I would glad to get any advice from you, how I can fix this problem myself. 
Thank you

**Comment by Ruenzuo** [comment]
Oh I understand.

Could you share the full output of the compiler command in a gist maybe? The error you shared isn't descriptive enough I'm afraid. I'd like to know if the linker is looking for an .o file or the filelist.

**Comment by mvyrko** [comment]
Please, take a look
[LinkerError.zip](https://github.com/Ruenzuo/cocoapods-amimono/files/630617/LinkerError.zip)

**Comment by Ruenzuo** [comment]
So you can start by verifying that the filelist is created correctly. You should find it in the following path: 

```
/Users/vyrkoma/Library/Developer/Xcode/DerivedData/iPhone-atflhkzcwofqigdumxsqzijadier/Build/Intermediates/Pods.build/Debug-iphonesimulator-x86_64.objects.filelist
```

Just look for `-filelist` param on the linker command (might change if you clean DerivedData I believe). If the file it's there, check if it contains all libraries your application should link to. That might be a good start.

**Comment by mvyrko** [comment]
As far as I see this file is empty.
